### PR TITLE
[BEAM-9184] Add ToSet combiner

### PR DESCRIPTION
**Pull request by sorensenjs**
Adds ToSet(), following from ToList() and ToDict(). For lists with many duplicates, ToList() will not combine well. Pairing with an empty value to use ToDict() seems unnecessarily complicated.

### CODE REVIEW

**Comment by tvalentyn** [comment]
LGTM, thank you!
cc: @robertwb

File: sdks/python/apache_beam/transforms/combiners.py [hunk 1]
```diff
@@ -802,6 +804,35 @@ def extract_output(self, accumulator):
     return accumulator
 
 
+class ToSet(ptransform.PTransform):
+  """A global CombineFn that condenses a PCollection into a set."""
+
+  def __init__(self, label='ToSet'):  # pylint: disable=useless-super-delegation
```
Comment (tvalentyn) [inline sdks/python/apache_beam/transforms/combiners.py]: Is this pylint directive still required with the version of pylint Beam is using today?

Comment (sorensenjs) [reply sdks/python/apache_beam/transforms/combiners.py]: Copied from line 775 (now line 777). Do you want me to experiment? A change removing this seems like it should be a separate change.

Comment (sorensenjs) [reply sdks/python/apache_beam/transforms/combiners.py]: Removing this produces
************* Module apache_beam.transforms.combiners
apache_beam/transforms/combiners.py:810:2: W0235: Useless super delegation in method '__init__' (useless-super-delegation)

------------------------------------

Comment (tvalentyn) [reply sdks/python/apache_beam/transforms/combiners.py]: Thanks for checking.

**Comment by tvalentyn** [comment]
Checked that test failure doesn't happen locally, rerunning the suite.

**Comment by tvalentyn** [comment]
Run Python PreCommit

**Comment by tvalentyn** [comment]
Run Python PreCommit

File: sdks/python/apache_beam/transforms/combiners_test.py [hunk 2]
```diff
@@ -346,6 +346,18 @@ def match(actual):
     assert_that(result, matcher())
     pipeline.run()
 
+  def test_to_set(self):
+    pipeline = TestPipeline()
+    the_list = [6, 3, 1, 1, 9, 1, 5, 2, 0, 6]
+    pcoll = pipeline | 'start' >> Create(the_list)
+    result = pcoll | 'to list' >> combine.ToSet()
```
Comment (EDjur) [inline sdks/python/apache_beam/transforms/combiners_test.py]: Saw this in JIRA and made the exact same implementation before I realised the PR was already out :)

Small comment: there's a tiny typo here that doesn't really matter. Should be 'to set'.

**Comment by tvalentyn** [comment]
Run Python PreCommit

**Comment by tvalentyn** [comment]
Run Python PreCommit

**Comment by tvalentyn** [comment]
Run PythonLint PreCommit

**Comment by tvalentyn** [comment]
Run PythonLint PreCommit

**Comment by tvalentyn** [comment]
https://github.com/apache/beam/pull/10688 should fix the flaky test.

**Comment by tvalentyn** [comment]
Run Python PreCommit

**Pull request merged**